### PR TITLE
Delete user.

### PR DESCRIPTION
**Pull request by nicofecioru**

### CODE REVIEW

File: src/main/java/meeseeks/box/repository/UserRepository.java [hunk 1]
```diff
@@ -15,4 +18,9 @@
 public interface UserRepository extends CrudRepository<UserEntity, Integer> {
     Optional<UserEntity> findByUsername(final String username);
     Optional<UserEntity> findByEmail(final String email);
+
+    @Transactional
+    @Modifying
+    @Query("delete from UserEntity user where user.username = ?1")
+    int deleteUser(final String userName);
```
Comment (alexandrustoica) [inline src/main/java/meeseeks/box/repository/UserRepository.java]: This should return Long (but will convert it to int)

File: src/main/java/meeseeks/box/controller/UserController.java [hunk 2]
```diff
@@ -44,4 +49,12 @@ public void changePassword(Authentication auth, @RequestBody @Valid ChangePasswo
         user.setPassword(model.getPassword());
         userService.saveUser(user);
     }
+
+    @RequestMapping("/delete")
+    public ResponseEntity delete() {
+        UserEntity user =(UserEntity) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        return userService.getUserRepository().deleteUser(user.getUsername()) > 0 ?
```
Comment (alexandrustoica) [inline src/main/java/meeseeks/box/controller/UserController.java]: Don't 'get' the userRepository, make an autowired attr. in our controller. -> and use that in your tests

Comment (alexandrustoica) [reply src/main/java/meeseeks/box/controller/UserController.java]: Put in UserController constructor [final UserRepository userRepository]. Spring will inject that into your controller. 

userService.getUserRepository() will reference the same injected userRepo from this class, so you can remove the function getUserRepository() from UserService

Comment (decebaldecebal) [reply src/main/java/meeseeks/box/controller/UserController.java]: Please change this.

File: src/main/java/meeseeks/box/repository/UserRepository.java (hunk 1 above)
Comment (decebaldecebal) [reply src/main/java/meeseeks/box/repository/UserRepository.java]: Apparently long throws an error.

**Pull request merged**